pymeasure/instruments/toptica/ibeamsmart.py

```diff
@@ -22,99 +22,267 @@
+
+    .. code::
+
+        laser = IBeamSmart("SomeResourceString")
+        laser.emission = True
+        laser.ch_2.power = 1000  # µW
+        laser.ch_2.enabled = True
+        laser.shutdown()
+
+    :param adapter: pyvisa resource name or adapter instance.
+    :param \\**kwargs: Any valid key-word argument for VISAAdapter.
     """
-    def __init__(self, adapter, baud_rate=115200, **kwargs):
-        if not isinstance(adapter, str):
-            raise TypeError("'adapter' must be a pyvisa resource name string")
+    _reg_value = re.compile(r"\w+\s+=\s+(\w+)")
+
+    channels = Instrument.ChannelCreator(DriverChannel, (1, 2, 3, 4, 5))
+
+    def __init__(self, adapter, name="Toptica IBeam Smart laser diode", **kwargs):
         super().__init__(
-            TopticaAdapter(adapter, baud_rate, **kwargs),
-            "toptica IBeam Smart laser diode",
+            adapter,
+            name,
             includeSCPI=False,
+            read_termination='\r\n',
+            write_termination='\r\n',
+            baud_rate=115200,
+            **kwargs
         )
+        # configure communication mode: no repeating and no command prompt
+        self.write('echo off')
+        self.write('prom off')
+        time.sleep(0.04)
+        # clear the initial messages from the controller
+        try:
+            self.adapter.flush_read_buffer()
+        except AttributeError:
+            log.warning("Adapter does not have 'flush_read_buffer' method.")
+        self.ask('talk usual')
+
+    def read(self):
+        """Read a reply of the instrument and exctract the values, if possible.
+
+        Reads a reply of the instrument which consists of at least two
+        lines. The initial ones are the reply to the command while the last one
+        should be '[OK]' which acknowledges that the device is ready to receive
+        more commands.
+
+        Note: '[OK]' is always returned as last message even in case of an
+        invalid command, where a message indicating the error is returned
+        before the '[OK]'
+
+        Value extraction: extract <value> from 'name = <value> [unit]'.
+        If <value> can not be identified the orignal string is returned.
+
+        :return: string containing the ASCII response of the instrument (without '[OK]').
+        """
+        reply = super().read()  # read back the LF+CR which is always sent back
+        if reply != "":
+            raise ValueError(
+                "TopticaAdapter.read(1): Error after command "
+                f"'{self.lastcommand}' with message '{reply}'")
+        msg = []
+        try:
+            while True:
+                line = super().read()
+                if line == '[OK]':
+                    break
+                msg.append(line)
+        except VisaIOError:
+            reply = '\n'.join(msg)
+            try:
+                self.adapter.connection.flush_read_buffer()
+            except AttributeError:
```

Comment:
Same as in line 108. NotImplementedError should likely be caught and a fallback for the not existing flush method would be needed

Reply:
See above.